recipes/hyperscan/all/conanfile.py

```diff
@@ -0,0 +1,96 @@
+from conans import ConanFile, CMake, tools
+
+import os
+
+class HyperscanConan(ConanFile):
+    name = "hyperscan"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.hyperscan.io"
+    description = "High-performance regular expression matching library"
+    topics = ("regex", "regular expressions")
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+
+    _cmake = None
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "optimise": [True, False],
+        "debug_output": [True, False],
```

Comment:
Is this really required?

https://github.com/intel/hyperscan/blob/64a995bf445d86b74eb0f375624ffc85682eadfe/doc/dev-reference/getting_started.rst#cmake-configuration

Reply:
It's an option provided by upstream. Right now all it does is to disable optimizations and define the DEBUG macro (I'm not really fit with CMake, but I assume that's basically what the debug build type does as well). However, I feel it's good practice to expose this option to the conan recipe, since we can never know if upstream will do some more magic in a future version when this option is selected (as opposed to only specifying the CMake build type). That said, that's my first contribution to CCI and I don't know if there are any guidelines about such questions.

Reply:
What if the consumer does a Release build (`conan install .. -s build_type=Release`) and has a nasty surprise that it's actually a debug version?

Usually adding all the options is the right choice but my concern is it might be too much...

> ❗ This is just a an opinion and you are not obliged to follow (the internet is a free place)

If you had you're heart set on offering these you could add an "auto" and use that as default...

```py
if self.settings.debug_output != "auto":
    cmake.def[...] = self.settings.debug_output
```

This way the behavior matches upstream but allows users to override if they wish

Reply:
I'm not heart set. Anyway, I used the approach with auto options for now. Grepping the other recipes however showed that this is actually really rare. I'm fine with removing the optimise, debug_output and dump_support options completely if that is regarded as the cleaner solution.

Reply:
That's because it's not a common practice in C++ projects... it leads to inconsistencies on different platforms which are always a pain to manage....

Let's see if anyone else chimes in other wise it's fine!